src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,213 @@
+package frc.trigon.robot.subsystems.arm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.*;
+import frc.trigon.robot.utilities.Conversions;
+import java.util.Set;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+
+    private final CANSparkMax
+            masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile.Constraints
+            angleConstraints = ArmConstants.ANGLE_CONSTRAINTS,
+            elevatorConstraints = ArmConstants.ELEVATOR_CONSTRAINTS;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+    }
+
+    public Command getSetTargetArmStateCommand(ArmConstants.ArmState targetState) {
+        return setTargetArmPosition(targetState.angle, targetState.elevatorPosition, 100, 100);
+    }
+
+    public Command getSetTargetArmPositionCommand(Rotation2d angle, double elevatorPosition) {
+        return setTargetArmPosition(angle, elevatorPosition, 100, 100);
+    }
+
+    public Command getSetTargetArmPositionCommandAndSetPercentage(ArmConstants.ArmState targetState, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        return setTargetArmPosition(targetState.angle, targetState.elevatorPosition, angleSpeedPercentage, elevatorSpeedPercentage);
+    }
+
+    private Command setTargetArmPosition(Rotation2d targetAngle, double targetElevatorPosition, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        angleConstraints = Conversions.scaleConstraints(ArmConstants.ANGLE_CONSTRAINTS, angleSpeedPercentage);
+        elevatorConstraints = Conversions.scaleConstraints(ArmConstants.ELEVATOR_CONSTRAINTS, elevatorSpeedPercentage);
```

Comment:
though this would work, there's a more logical and efficient way to do this. Try passing the speed percentage as a param to the profile generation function.